flexget/plugins/input/npo_watchlist.py

```diff
@@ -95,173 +106,151 @@ def _login(self, task, config):
 
             if 'isAuthenticatedUser' not in profile_response.cookies:
                 raise plugin.PluginError('Failed to login. Check username and password.')
+            log.debug('Succesfully logged in: %s', email)
         except RequestException as e:
             raise plugin.PluginError('Request error: %s' % e.args[0])
 
-    def _get_page(self, task, config, url):
-        self._login(task, config)
+    def _get_favourites_entries(self, task, config, profileId):
+        # Details of profile using the $profileId:
+        # https://www.npostart.nl/api/account/@me/profile/$profileId  (?refresh=1)
+        # XMLHttpRequest could also be done on, but is not reliable, because if a series is
+        # bookmarked after episode is already broadcasted, it does not appear in this list.
+        # https://www.npostart.nl/ums/accounts/@me/favourites/episodes?dateFrom=2018-06-04&dateTo=2018-06-05
+        profile_details_url = 'https://www.npostart.nl/api/account/@me/profile/'
 
+        entries = []
         try:
-            page_response = requests.get(url)
-            if page_response.url != url:
-                raise plugin.PluginError('Unexpected page: {} (expected {})'.format(page_response.url, url))
-
-            return page_response
+            profile = requests.get(profile_details_url + profileId).json()
+            log.info('Retrieving favorites for profile %s', profile['name'])
```

Comment:
level

Reply:
I think I'd like to prefer this one as an 'info' so that the user can see which profile (on the new NPO site there can be multiple) within their account is used; they can't choose a profile within Flexget, so it is important to know which one is actually used.

Reply:
Info log should describe an action that occurs once per flow. Is this the case here?

Reply:
Yes, this just once per flow; it returns the name of the profile that will be used. From this profile the series that are marked as 'favorite' are checked for new episodes.